src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machineProducer;
+        if (type.equals(Excavator.class)) {
+            machineProducer = new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            machineProducer = new TruckProducer().get();
+        } else if (type.equals(Bulldozer.class)) {
+            machineProducer = new BulldozerProducer().get();
+        } else {
+            machineProducer = new ArrayList<>();
+        }
+        return new ArrayList<>(machineProducer);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. The variable `machineProducer` is already declared, but it's better to use a more descriptive name and directly return the result without creating a new ArrayList, unless you have a specific reason for copying the list.